src/reanimated2/Hooks.js

```diff
@@ -47,7 +47,15 @@ export function useEvent(handler, eventNames = [], rebuild = false) {
   } else if (rebuild) {
     initRef.current.updateWorklet(handler);
   }
-  return initRef.current;
+
+  useEffect(() => {
+    return () => {
+      initRef.current.unregisterFromEvents();
```

Comment:
what's the point of using ref when it's being trashed on unmount? I thought the refs are used because we want the objects to persist between mounts